integration/ig/k8s/top_file_test.go

```diff
@@ -23,49 +23,88 @@ import (
 	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
+func newTopFileCmd(ns string, cmd string, startAndStop bool) *Command {
+	expectedOutputFn := func(output string) error {
+		expectedEntry := &types.Stats{
+			CommonData: eventtypes.CommonData{
+				Namespace: ns,
+				Pod:       "test-pod",
+			},
+			// echo is built-in
+			Comm:     "sh",
+			Filename: "bar",
+			FileType: 'R',
+		}
+
+		normalize := func(e *types.Stats) {
+			e.Container = ""
+			e.Pid = 0
+			e.Tid = 0
+			e.MountNsID = 0
+			e.Reads = 0
+			e.ReadBytes = 0
+			e.Writes = 0
+			e.WriteBytes = 0
+		}
+
+		return ExpectEntriesInMultipleArrayToMatch(output, normalize, expectedEntry)
+	}
+
+	return &Command{
+		Name:             "TopFile",
+		ExpectedOutputFn: expectedOutputFn,
+		Cmd:              cmd,
+		StartAndStop:     startAndStop,
+	}
+}
+
 func TestTopFile(t *testing.T) {
 	t.Parallel()
 	ns := GenerateTestNamespaceName("test-top-file")
 
-	topFileCmd := &Command{
-		Name:         "TopFile",
-		Cmd:          fmt.Sprintf("ig top file -o json -m 999 --runtimes=%s", *containerRuntime),
-		StartAndStop: true,
-		ExpectedOutputFn: func(output string) error {
-			expectedEntry := &types.Stats{
-				CommonData: eventtypes.CommonData{
-					Namespace: ns,
-					Pod:       "test-pod",
-				},
-				// echo is built-in
-				Comm:     "sh",
-				Filename: "bar",
-				FileType: 'R',
-			}
-
-			normalize := func(e *types.Stats) {
-				e.Container = ""
-				e.Pid = 0
-				e.Tid = 0
-				e.MountNsID = 0
-				e.Reads = 0
-				e.ReadBytes = 0
-				e.Writes = 0
-				e.WriteBytes = 0
-			}
-
-			return ExpectEntriesInMultipleArrayToMatch(output, normalize, expectedEntry)
-		},
-	}
-
-	commands := []*Command{
+	commandsPreTest := []*Command{
 		CreateTestNamespaceCommand(ns),
-		topFileCmd,
-		SleepForSecondsCommand(2), // wait to ensure ig has started
```

Comment:
Don't we need this anymore?

Reply:
We don't because we are now running `ig` after creating the pod.